### PR TITLE
optimize local thumb grabbing

### PR DESCRIPTION
**Pull request by NadavHarnik**
- [x] add packager URL to config

- [x] add location to packager config

### CODE REVIEW

File: alpha/apps/kaltura/lib/myEntryUtils.class.php [hunk 1]
```diff
@@ -895,7 +908,57 @@ public static function captureLocalThumb($entry, $capturedThumbPath, $calc_vid_s
...
+			return false;
+
+		//look for the highest bitrate flavor tagged with thumbsource
+		$flavorAsset = assetPeer::retrieveHighestBitrateByEntryId($entry->getId(), flavorParams::TAG_THUMBSOURCE);
+
+		if(is_null($flavorAsset) || !self::isFlavorSupportedByPackager($flavorAsset))
+		{
+			// look for the highest bitrate flavor the packager can parse
+			$flavorAsset = assetPeer::retrieveHighestBitrateByEntryId($entry->getId(), flavorParams::TAG_MBR);
+			if (is_null($flavorAsset))
```
Comment (noam-arad) [inline alpha/apps/kaltura/lib/myEntryUtils.class.php]: shouldn't you also check here if $flavorAsset is not supported by packager?

File: alpha/apps/kaltura/lib/myEntryUtils.class.php [hunk 2]
```diff
@@ -895,7 +908,57 @@ public static function captureLocalThumb($entry, $capturedThumbPath, $calc_vid_s
...
+		$success = KCurlWrapper::getDataFromFile($packagerThumbCapture, $tempThumbPath);
+		if(!$success)
+			return false;
+
+		return true;
+	}
+
+	public static function isFlavorSupportedByPackager($flavorAsset)
+	{
+		if(($flavorAsset->hasTag(flavorParams::TAG_WEB) && in_array($flavorAsset->getContainerFormat(), array('mp42','isom','f4v'))))
```
Comment (noam-arad) [inline alpha/apps/kaltura/lib/myEntryUtils.class.php]: move constant strings to const

File: alpha/apps/kaltura/lib/myEntryUtils.class.php [hunk 3]
```diff
@@ -895,7 +908,57 @@ public static function captureLocalThumb($entry, $capturedThumbPath, $calc_vid_s
...
+			{
+				//retrieve original ready
+				$flavorAsset = assetPeer::retrieveOriginalReadyByEntryId($entry->getId());
+				if(is_null($flavorAsset) || !self::isFlavorSupportedByPackager($flavorAsset))
+					return false;
+			}
+		}
+
+		$flavorAssetId = $flavorAsset->getId();
+		$flavorSyncKey = $flavorAsset->getSyncKey(flavorAsset::FILE_SYNC_ASSET_SUB_TYPE_ASSET);
```
Comment (MosheMaorKaltura) [inline alpha/apps/kaltura/lib/myEntryUtils.class.php]: should be file sync key

File: alpha/apps/kaltura/lib/myEntryUtils.class.php [hunk 4]
```diff
@@ -895,7 +908,57 @@ public static function captureLocalThumb($entry, $capturedThumbPath, $calc_vid_s
...
+			// look for the highest bitrate flavor the packager can parse
+			$flavorAsset = assetPeer::retrieveHighestBitrateByEntryId($entry->getId(), flavorParams::TAG_MBR);
+			if (is_null($flavorAsset))
+			{
+				//retrieve original ready
+				$flavorAsset = assetPeer::retrieveOriginalReadyByEntryId($entry->getId());
+				if(is_null($flavorAsset) || !self::isFlavorSupportedByPackager($flavorAsset))
+					return false;
+			}
+		}
```
Comment (MosheMaorKaltura) [inline alpha/apps/kaltura/lib/myEntryUtils.class.php]: Shoud be a private function that returns flavorAsset that is supported by packager

File: alpha/apps/kaltura/lib/myEntryUtils.class.php [hunk 5]
```diff
@@ -895,7 +908,57 @@ public static function captureLocalThumb($entry, $capturedThumbPath, $calc_vid_s
...
+
+		if (!$entry_data_path)
+			return false;
+
+		$packagerThumbCapture = str_replace(
+			array ( "{url}", "{offset}" ),
+			array ( $entry_data_path , floor($calc_vid_sec*1000)  ) ,
+			$packagerCaptureUrl );
+
+		$tempThumbPath = $capturedThumbPath.'temp_1.jpg';
```
Comment (MosheMaorKaltura) [inline alpha/apps/kaltura/lib/myEntryUtils.class.php]: Use const temp_1.jpg CAPTURE_THUMB_TEMP name

**Pull request merged**